vllm/v1/engine/input_processor.py

```diff
@@ -403,14 +403,19 @@ def _extract_mm_data(p: PromptType):
                 if isinstance(data, list) or MultiModalDataParser.is_embeddings(data)
                 else 1
             )
-            mm_uuids[modality] = [f"{request_id}-{modality}-{i}" for i in range(n)]
+            # request_id is externally provided, add randomness to ensure uniqueness
+            mm_uuids[modality] = [
+                f"{request_id}-{modality}-{random_uuid():.8}-{i}" for i in range(n)
+            ]
         return mm_uuids
 
-    def _generate_request_id(self, request_id: str):
-        """Construct an internal request ID by adding 8 random characters
-        to the supplied request ID in order to ensure uniquness.
+    def assign_request_id(self, request: EngineCoreRequest):
+        """Replace the externally supplied request ID with an internal request ID
+        that adds 8 random characters in order to ensure uniquness.
         """
-        return f"{request_id}-{random_uuid():.8}"
+        assert request.external_req_id is None
```

Comment:
This should probably be changed to raise a runtime exception. It should be ok to fail the request in this case rather than crash the server.